ndnsd/communication/sync-adapter.cpp

```diff
@@ -15,28 +15,31 @@
  *
  * You should have received a copy of the GNU Lesser General Public License along with
  * NDNSD, e.g., in COPYING.md file.  If not, see <http://www.gnu.org/licenses/>.
+
+  @ most part of sync-adapter code is taken from NLSR/communication/
+
  **/
 
 #include "sync-adapter.hpp"
-// #include "logger.hpp"
+#include <iostream>
 
-// INIT_LOGGER(SyncProtocolAdapter);
+INIT_LOGGER(SyncProtocolAdapter);
 
 namespace ndnsd {
 
 const auto FIXED_SESSION = ndn::name::Component::fromNumber(0);
```

Comment:
So you are also using fixed session for ChronoSync. Hope you know why NLSR uses it and this applies in this context too. Of course PSync does not provide session functionality. Maybe rename this CHRONOSYNC_FIXED_SESSION

Reply:
The prefixes are tied to a particular session. If a session is destroyed all the associated prefix will also get destroyed/cleaned?